iarc_main/scripts/follow.py

```diff
@@ -24,9 +25,20 @@ def run(self):
 
 	            # Plan the motion
 	            cmd = Twist()
-	            cmd.linear.x = position[0]
-	            cmd.linear.y = position[1]
-	            cmd.angular.z = 0
+
+	            # maxvelocity of our drone
+	            maxvelocity = 1
+
+	            # diagonalvelocity = the velocity we would get if we use position[0] and position[1]
+	            diagonalvelocity = math.sqrt(position[0]**2 + position[1]**2)
+
+	            # Scale down our velocity_x and velocity_y so that our diagonalvelocity won't exceed maxvelocity
+	            # But if diagonalvelocity is already < max_velocity then use diagonalvelocity instead (slow down when approach nearer)
+	            cmd.linear.x = position[0] if diagonalvelocity < maxvelocity else position[0] / diagonalvelocity * maxvelocity
+	            cmd.linear.y = position[1] if diagonalvelocity < maxvelocity else position[1] / diagonalvelocity * maxvelocity
+	            
+	            # Turn our drone to the position_x/y of the target
+	            cmd.angular.z = math.atan2(position[1], position[0])
```

Comment:
It may be worth having an "aggressiveness" control that multiplies the angle difference to get a desired angular velocity. Using 1 seems to work well in simulation, but it could be inappropriate for a real drone.